graph.py

```diff
@@ -0,0 +1,47 @@
+import random
+
+
+class Graph:
+    def __init__(self):
+        """
+        Graph variable is a dictionary of lists - graph is implemented as adjacency lists
+        """
+        self.graph = dict()
+
+    def _topological_sort(self, root, visited, result):
+        visited.add(root)
+        print(root)
+        for neighbour in self.graph[root]:
+            if neighbour not in visited:
+                result, visited = self._topological_sort(neighbour, visited, result)
+
+        result.append(root)
+
+        return result, visited
+
+    def start_topological_sort(self):
+        """
+        We perform topological sort for each node, if it was not visited
+        :return list: contents of letters stack
+        """
+        visited = set()
+        result = list()
+        nodes = list(self.graph.keys())
+
+        if not nodes:
+            return []
+
+        for n in nodes:
+            if n not in visited:
+                result, visited = self._topological_sort(n, visited, result)
+
+        return result[::-1]
```

Comment:
There is also `reversed` that does the same thing (though this is fine too):

https://docs.python.org/3/library/functions.html#reversed